plugins/bundle/plugin.go

```diff
@@ -152,12 +153,12 @@ func (p *Plugin) Reconfigure(ctx context.Context, config interface{}) {
 	// Use p.cfgMtx instead of p.mtx so as to not block any bundle downloads/activations
 	// that are in progress. We upgrade to p.mtx locking after stopping downloaders.
 	p.cfgMtx.Lock()
-	defer p.cfgMtx.Unlock()
 
 	// Look for any bundles that have had their config changed, are new, or have been removed
 	newConfig := config.(*Config)
 	newBundles, updatedBundles, deletedBundles := p.configDelta(newConfig)
 	p.config = *newConfig
+	p.cfgMtx.Unlock()
```

Comment:
Have you considered the scenario of concurrent calls to `Reconfigure`? Could it break the following code flow?

What if we keep the existing code here?